libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -272,6 +272,23 @@ void AP_AIS::send(mavlink_channel_t chan)
         }
     }
 }
+void AP_AIS::handle_message(const mavlink_channel_t chan, const mavlink_message_t &msg)
+{
+    if (msg.msgid != MAVLINK_MSG_ID_AIS_VESSEL) {
+        return;
+    }
+
+    mavlink_ais_vessel_t packet;
+    mavlink_msg_ais_vessel_decode(&msg, &packet);
+
+    uint16_t index;
+    if (!get_vessel_index(packet.MMSI, index)) {
+        // not found and not enough memory to add it
+        return;
+    }
+    memcpy(&_list[index].info, &packet, sizeof(mavlink_ais_vessel_t));
```

Comment:
```suggestion
    memcpy(&_list[index].info, &packet, _list[index].info);
```

Reply:
```suggestion
    memcpy(&_list[index].info, &packet, sizeof(_list[index].info));
```
I guess @peterbarker meant this.

Reply:
yup, changed.